pinot-integration-tests/src/test/java/org/apache/pinot/compat/tests/QueryOp.java

```diff
@@ -55,7 +72,97 @@ public void setExpectedResultsFileName(String expectedResultsFileName) {
 
   @Override
   boolean runOp() {
-    System.out.println("Verifying queries in " + _queryFileName + " against results in " + _expectedResultsFileName);
-    return true;
+    System.out.printf("Verifying queries in %s against results in %s\n", _queryFileName, _expectedResultsFileName);
+    try {
+      return verifyQueries();
+    } catch (Exception e) {
+      LOGGER.error("FAILED to verify queries in {}: {}", _queryFileName, e);
+      return false;
+    }
+  }
+
+  boolean verifyQueries()
+      throws Exception {
+    BufferedReader expectedResultReader = null;
+    boolean testPassed = false;
+
+    try (BufferedReader queryReader = new BufferedReader(
+        new InputStreamReader(new FileInputStream(_queryFileName), StandardCharsets.UTF_8))) {
+      if (_queryFileName == null) {
```

Comment:
I'm not sure whether this if statement can ever be satisfied. Maybe we should check that before the try block?

Reply:
Use a try-with-resources statement now.